src/transaction/payment.ts

```diff
@@ -85,29 +85,57 @@ function createMaximalAmount(amount: Amount): Amount {
   return _.assign({}, amount, {value: maxValue})
 }
 
+function normalizeAddress(address: string, tag: number | undefined, verification?: string):
+    {classicAddress: string, tag?: number} {
+  let addressClassic
+  if (isValidXAddress(address)) {
+    addressClassic = xAddressToClassicAddress(address)
+    addressClassic.tag = addressClassic.tag === false ? undefined : addressClassic.tag
+    if (tag !== undefined && addressClassic.tag !== tag) {
+      throw new ValidationError('address includes a tag that does not match the tag specified in the payment')
+    }
+  } else {
+    addressClassic = {
+      classicAddress: address,
+      tag
+    }
+  }
+
+  if (!verification) {
+    return addressClassic
+  }
+
+  let verificationClassic
```

Comment:
This is more or less the same logic logic as above. Maybe it makes sense to factor this to a helper? I'm not positive what verification is so it's a bit tough to say.